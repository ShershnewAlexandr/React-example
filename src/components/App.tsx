import React from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';
import { routes } from '../utils/constants';
import InformationPageContainer from '../containers/InformationPageContainer';
import LoginPage from './pages/loginPage/LoginPage';
import I18nHeaderContainer from '../containers/I18nHeaderContainer';
import CardListPageContainer from '../containers/CardListPageContainer';
import NewCardPageContainer from '../containers/newCard/NewCardPageContainer';
import EditCardPageContainer from '../containers/editCard/EditCardPageContainer';
import RegisterPage from './pages/registerPage/RegisterPage';

function App() {
  return (
    <div className="App">
      <I18nHeaderContainer />
      <Switch>
        <Route path={routes.SIGNIN} component={LoginPage} />
        <Route path={routes.INFORMATION} component={InformationPageContainer} />
        <Route path={routes.CARDLIST} component={CardListPageContainer} />
        <Route path={routes.NEWCARD} component={NewCardPageContainer} />
        <Route path={routes.EDITCARD} component={EditCardPageContainer} />
        <Route path={routes.REGISTER} component={RegisterPage} />
        <Route path="/" render={props => <Redirect to={routes.SIGNIN} />} />
      </Switch>
    </div>
  );
}

export default App;
