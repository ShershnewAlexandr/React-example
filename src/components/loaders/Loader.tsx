import React from 'react';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import loaderGif from '../../assets/img/loader.gif';

function Loader() {
  const { t } = useTranslation();
  return (
    <Grid
      item
      container
      direction="column"
      justify="center"
      alignItems="center"
    >
      <Grid item>
        <img className="loader" src={loaderGif} alt="Loader ..." />
      </Grid>
      <Grid item>
        <span>{t('loader')}</span>
      </Grid>
    </Grid>
  );
}

export default Loader;
