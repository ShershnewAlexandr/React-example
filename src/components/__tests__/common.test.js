import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import Loader from '../loaders/Loader';
import App from '../App';

jest.mock('../../store/index');
jest.mock('i18next', () => ({
  use: function() {
    return this;
  },
  init: jest.fn(),
}));
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('common', () => {
  const app = shallow(<App />);
  const loader = shallow(<Loader />);

  it('App snapshot', () => {
    expect(toJson(app)).toMatchSnapshot();
  });

  it('Loader snapshot', () => {
    expect(toJson(loader)).toMatchSnapshot();
  });
});
