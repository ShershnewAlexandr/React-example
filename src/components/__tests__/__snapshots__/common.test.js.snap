// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common App snapshot 1`] = `
<div
  className="App"
>
  <Connect(I18nHeaderContainer) />
  <Switch>
    <Route
      component={[Function]}
      path="/signin"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(InformationPageContainer)",
          "type": [Function],
        }
      }
      path="/information"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(CardListPageContainer)",
          "type": [Function],
        }
      }
      path="/cardlist"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(NewCardPageContainer)",
          "type": [Function],
        }
      }
      path="/newcard"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(EditCardPageContainer)",
          "type": [Function],
        }
      }
      path="/editcard/:id"
    />
    <Route
      component={[Function]}
      path="/register"
    />
    <Route
      path="/"
      render={[Function]}
    />
  </Switch>
</div>
`;

exports[`common Loader snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  item={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <img
      alt="Loader ..."
      className="loader"
      src="loader.gif"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <span>
      loader
    </span>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
